src/hera/workflows/runner.py

```diff
@@ -76,6 +83,17 @@ def _is_str_kwarg_of(key: str, f: Callable):
         return False
 
 
+def _map_keys(function: Callable, kwargs: dict):
+    """Change the kwargs's keys to use the python name instead of the parameter name which could be kebab case."""
+    mapped_kwargs = {}
```

Comment:
```suggestion
def _map_keys(function: Callable, kwargs: dict):
    """Change the kwargs's keys to use the python name instead of the parameter name which could be kebab case."""
    if not global_config.experimental_features["script_annotations"]:
        return {key.replace("-", "_"): value for key, value in kwargs.items()}

    mapped_kwargs = {}
```